compiler/src/dotty/tools/dotc/fromtasty/TastyFileUtil.scala

```diff
@@ -17,25 +18,20 @@ object TastyFileUtil {
    *    package foo
    *    class Foo
    *  ```
-   *  then `getClassName("./out/foo/Foo.tasty") returns `("./out", "foo.Foo")`
+   *  then `getClassName("./out/foo/Foo.tasty") returns `Some(("./out", "foo.Foo"))`
    */
-  def getClassName(path: Path): (String, String) = {
+  def getClassName(path: Path): Option[(String, String)] = {
     assert(path.toString.endsWith(".tasty"))
     assert(Files.exists(path))
     val bytes = Files.readAllBytes(path)
-    val unpickler: TastyUnpickler = new TastyUnpickler(bytes)
-    val className =
-      unpickler.nameAtRef.contents.iterator.dropWhile {
-        name => name.toString == TreePickler.sectionName || name == nme.EMPTY_PACKAGE
-      }.takeWhile {
-        case name: SimpleName => name != nme.CONSTRUCTOR
-        case name => !name.is(NameKinds.ModuleClassName)
-      }.collect {
-        case name: SimpleName => name.toString
-      }.toList
-    val classInPath = className.mkString("", "/", ".tasty")
-    val classpath = path.toString.replace(classInPath, "")
-    (classpath, className.mkString("."))
+    val names = new core.tasty.TastyClassName(bytes).readName()
+    names.map { case (packName, className) =>
+      val fullName = s"$packName.${className.lastPart}"
+      val classInPath = fullName.replace(".", "/") + ".tasty"
```

Comment:
Need to use `File.separator` instead of `/` as otherwise the replace fails on Windows and we get the full tasty file path as classpath.

Example:

```
[info] file:///c%3A/Users/Tobias/Programming/Scala/dotty/out/bootstrap/dotty-language-server/scala-0.11/classes/dotty/tools/languageserver/DottyLanguageServer.tasty
[info] classPath: c:\Users\Tobias\Programming\Scala\dotty\out\bootstrap\dotty-language-server\scala-0.11\classes\dotty\tools\languageserver\DottyLanguageServer.tasty
[info] className: dotty.tools.languageserver.DottyLanguageServer
[error] class dotty.tools.languageserver.DottyLanguageServer cannot be unpickled because no class file was found
```

Reply:
Changed to `File.separator`